Gems/ScriptCanvas/Code/Include/ScriptCanvas/Execution/Interpreted/ExecutionStateInterpreted.cpp

```diff
@@ -6,172 +6,60 @@
  *
  */
 
-#include <AzCore/Asset/AssetManager.h>
 #include <AzCore/Script/ScriptContext.h>
 #include <AzCore/Script/ScriptSystemBus.h>
 #include <Execution/Interpreted/ExecutionInterpretedAPI.h>
 #include <Execution/Interpreted/ExecutionStateInterpreted.h>
 #include <Execution/Interpreted/ExecutionStateInterpretedUtility.h>
 #include <Execution/RuntimeComponent.h>
 
-namespace ExecutionStateInterpretedCpp
-{
-    using namespace ScriptCanvas;
-
-    AZ::Data::Asset<RuntimeAsset> GetSubgraphAssetForDebug(const AZ::Data::AssetId& id)
-    {
-        // #functions2 this may have to be made recursive
-        auto asset = AZ::Data::AssetManager::Instance().GetAsset<SubgraphInterfaceAsset>(id, AZ::Data::AssetLoadBehavior::PreLoad);
-        asset.BlockUntilLoadComplete();
-        return asset;
-    }
-}
-
 namespace ScriptCanvas
 {
-    ExecutionStateInterpreted::ExecutionStateInterpreted(const ExecutionStateConfig& config)
+    ExecutionStateInterpreted::ExecutionStateInterpreted(ExecutionStateConfig& config)
         : ExecutionState(config)
         , m_interpretedAsset(config.runtimeData.m_script)
     {
-        RuntimeAsset* runtimeAsset = config.asset.Get();
+        RuntimeAsset* runtimeAsset = config.overrides.m_runtimeAsset.Get();
 
-#if defined(SCRIPT_CANVAS_RUNTIME_ASSET_CHECK)
+#if defined(SC_RUNTIME_CHECKS_ENABLED)
         if (!runtimeAsset)
         {
             AZ_Error("ScriptCanvas", false
                 , "ExecutionStateInterpreted created with ExecutionStateConfig that contained bad runtime asset data. %s"
-                , config.asset.GetId().ToString<AZStd::string>().data());
+                , config.overrides.m_runtimeAsset.GetId().ToString<AZStd::string>().data());
             return;
         }
 #else
         AZ_Assert(false
             , "ExecutionStateInterpreted created with ExecutionStateConfig that contained bad runtime asset data. %s"
-            , config.asset.GetId().ToString<AZStd::string>().data());
+            , config.overrides.m_runtimeAsset.GetId().ToString<AZStd::string>().data());
 #endif
 
-        if (!runtimeAsset->m_runtimeData.m_areStaticsInitialized)
+        if (!runtimeAsset->m_runtimeData.m_areScriptLocalStaticsInitialized)
         {
             Execution::InitializeInterpretedStatics(runtimeAsset->m_runtimeData);
         }
     }
 
-    void ExecutionStateInterpreted::ClearLuaRegistryIndex()
-    {
-        m_luaRegistryIndex = LUA_NOREF;
-    }
-
-    const Grammar::DebugExecution* ExecutionStateInterpreted::GetDebugSymbolIn(size_t index) const
-    {
-        return index < m_component->GetRuntimeAssetData().m_debugMap.m_ins.size()
-            ? &(m_component->GetRuntimeAssetData().m_debugMap.m_ins[index])
```

Comment:
this functionality was moved to the base execution state class, as it would be the same no matter how SC runtime is compiled.